ansible/pbTestScripts/buildJDK.sh

```diff
@@ -6,13 +6,18 @@ if [ -z "${WORKSPACE:-}" ]; then
 		WORKSPACE=$HOME
 fi
 [[ ! -d "$WORKSPACE/openjdk-build" ]] && git clone https://github.com/adoptopenjdk/openjdk-build $WORKSPACE/openjdk-build
+
+# If detected GCC version is <7, set it to 7 
+export gccVer=$(gcc --version | grep "gcc" | awk '{ print $4 }' | cut -d. -f1)
+[[ ${gccVer} -lt 7 ]] && ln -sf /usr/bin/gcc-7 /usr/bin/gcc && ln -sf /usr/bin/g++-7 /usr/bin/g++
+
 export TARGET_OS=linux
 export ARCHITECTURE=x64
 export JAVA_TO_BUILD=jdk8u
 export VARIANT=openj9
 export JDK7_BOOT_DIR=/usr/lib/jvm/java-1.7.0
```

Comment:
Are the packages with the OS really different in their naming formats between java 7 and 8?

Reply:
Looking at C6, U16 and U18, it seems they also have different names for Java 7:
C6: `java-1.7.0 `**or** `java-1.7.0-openjdk-1.7.0.241.x86_64`
U16: `java-1.7.0-openjdk-amd64` **or** `java-7-openjdk-amd64`
U18: No Java 7 installed. Probably due to `openjdk-7-jdk` only being under the Ubuntu1604 list of extra packages to install: https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/29ce5a65c94510fd91dcd29c42dd07ffcb3ff006/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/Ubuntu.yml#L72

Reply:
If they all have a "well-defined" symlink name, then perhaps test for each possibility (i.e. two of them - `java-1.7.0` or `java-1.7.0-openjdk-amd64`) and set it accordingly?